pkg/reconcile/reconcile.go

```diff
@@ -91,3 +93,26 @@ var _ Reconciler = Func(nil)
 
 // Reconcile implements Reconciler.
 func (r Func) Reconcile(o Request) (Result, error) { return r(o) }
+
+// PanicHandler can be called with defer in order to recover from panic in reconcile logic.
+// Additional handlers can also be passed to manage the go routines. The error passed to the
+// PanicHandler is set to the value of panic if not nil.
+// If any resource triggers a panic, the reconciler will recover and modify the raised panic
+// to the normal reconciler error.
+// Example usage:
+/*
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (result reconcile.Result, err error) {
+	defer HandleCrashReconciler(&err)
+	// reconciler logic here
+
+	...
+	return reconcile.Result{}, nil
+*/
+func PanicHandler(err *error, panicHandlers ...func(interface{})) {
```

Comment:
`PanicHandler` is not tied with the implementation of `Renconciler` and can be an independent helper function, having it here since it was targeted to use with reconciler logic.